File: src/main/java/com/semiprj/data/Room.java
package com.semiprj.data;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.util.HashMap;
import java.util.Map;


@Getter
@Setter
@ToString
public class Room {
    public Room(){}

    private String roomUuid;
    private CamClient hostClient;
    private Map<Integer, CamClient> camClients = new HashMap<>();
    private int lastModified;
    private int maxUser = 4;

    public boolean addClient(CamClient client){
        int index = ++lastModified;
        client.setIndex(index);
        camClients.put(index, client);

        return true;
    }

    public boolean removeClient(CamClient client){
        camClients.remove(client.getIndex());
        ++this.lastModified;

        return true;
    }
}

File: src/main/webapp/resources/frame/data/Room.js

export default class Room{
    constructor(){
        this._roomUuid;
        this._hostClient;
        this._camClients = new Map();
        this._lastModified = -1;
        this._maxUser;
    }

    spawn(hostClient){
        this._roomUuid = crypto.randomUUID();
        this._hostClient = hostClient;
        this.addClient(hostClient);
        return this;
    }

    addClient(client){
        const index = ++this._lastModified;
        client.index = index;
        this._camClients.set(index, client);

        return true;
    }

    removeClient(client){
        this._camClients.delete(client.index);
        ++this._lastModified;

        return true;
    }

    get roomUuid() {
        return this._roomUuid;
    }

    get lastModified(){
        return this._lastModified;
    }

    get camClients(){
        return this._camClients;
    }

    get maxUser(){
        return this._maxUser;
    }
}
